### PR TITLE
Try using clang-tidy-review action to check PR code

### PR DESCRIPTION
**Pull request by ghutchis**
Adapted from Mudlet
https://github.com/Mudlet/Mudlet/blob/development/.github/workflows/clangtidy-diff-analysis.yml

Signed-off-by: Geoff Hutchison <geoff.hutchison@gmail.com>

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Comment by ghutchis** [comment]
This can't run in this style (i.e. from user's forks) due to this issue:
https://github.com/actions/first-interaction/issues/10

**Comment by ghutchis** [comment]
I'm going to look into a workflow-run version of the script (e.g, saving the compile commands JSON somewhere).

**Comment by ghutchis** [comment]
It's basically impossible to run and post as comments with our workflow (pull request from a user's fork).

I'll put this off and rewrite to just run clang-tidy and leave any comments / warnings as a fail.

File: .github/workflows/build_cmake.yml [hunk 1]
```diff
@@ -22,6 +22,13 @@ jobs:
       fail-fast: false
       matrix:
         config:
+        - {
+            name: "Ubuntu Latest Clang", artifact: "",
+            os: ubuntu-latest,
+            cc: "clang", cxx: "clang++",
+            cmake_flags: "-G Ninja -DENABLE_TESTING=ON -DCMAKE_EXPORT_COMPILE_COMMANDS=1",
```
Comment (e-kwsm) [inline .github/workflows/build_cmake.yml:29]: CMake 3.6 or later supports [`CMAKE_CXX_CLANG_TIDY`](https://cmake.org/cmake/help/v3.6/variable/CMAKE_LANG_CLANG_TIDY.html) variable.
With `-DCMAKE_CXX_CLANG_TIDY=clang-tidy;--fix`, clang-tidy fixes source files if possible.

Comment (ghutchis) [reply .github/workflows/build_cmake.yml:29]: Today Iearned .. 😄

Comment (e-kwsm) [reply .github/workflows/build_cmake.yml:29]: Note that multiple build processes launched by ninja may modify a header file at the same time, which messes the file and results in build failure.

Comment (e-kwsm) [reply .github/workflows/build_cmake.yml:29]: So, the serial script would be preferable to avoid a spurious build failure.

**Comment by ghutchis** [comment]
I'm closing this in favor of a simpler approach...

**Pull request closed without merging**